benchmarks/segmentation_layers.py

```diff
@@ -1,175 +1,275 @@
-    nt = nestedtensor.nested_tensor(inputs)
-    batch_norm = torch.nn.BatchNorm2d(30, 1e-05, 0.1)
-    batch_norm.eval()
-
-    def _batch_norm_nt():
-        batch_norm(nt)
-    
-    return _batch_norm_nt
-
-#
-# max_pool2d
-#
-def max_pool2d_tensor_tensors_same_size():
-    inputs = [torch.randn(C, H, W) for _ in range(N)]
-    a = torch.stack(inputs)
-    maxPool2d = torch.nn.MaxPool2d(30)
-
-    def _max_pool2d_tensor():
-        maxPool2d(a)
-    return _max_pool2d_tensor
-
-def max_pool2d_nt_tensors_same_size():
-    inputs = [torch.randn(C, H, W) for _ in range(N)]
-    maxPool2d = torch.nn.MaxPool2d(30)
-    nt = nestedtensor.nested_tensor(inputs)
-    def _max_pool2d_nt():
-        maxPool2d(nt)
-
-    return _max_pool2d_nt
-
-#
-# cross_entropy
-#
-def cross_entropy_tensor_tensors_same_size():
-    inputs = [torch.randn(C, H, W) for _ in range(N)]
-    targets = [torch.randint(1, (H, W), dtype=torch.int64) for _ in range(N)]
-    
-    a = torch.stack(inputs)
-    b = torch.stack(targets)
-    
-
-    def _cross_entropy_tensor():
-        torch.nn.functional.cross_entropy(a, b)
-    return _cross_entropy_tensor
-
-def cross_entropy_nt_tensors_same_size():
-    inputs = [torch.randn(C, H, W) for _ in range(N)]
-    targets = [torch.randint(1, (H, W), dtype=torch.int64) for _ in range(N)]
-
-    a = nestedtensor.nested_tensor(inputs)
-    b = nestedtensor.nested_tensor(targets)
-
-    def _cross_entropy_nt():
-        torch.nn.functional.cross_entropy(a, b)
-    return _cross_entropy_nt
-
-#
-# dropout
-#
-def dropout_tensor_tensors_same_size():
-    inputs = [torch.randn(C, H, W) for _ in range(N)]
-    a = torch.stack(inputs)
-
-    def _dropout_tensor():
-        torch.nn.functional.dropout(a)
-    return _dropout_tensor
-
-def dropout_nt_tensors_same_size():
-    inputs = [torch.randn(C, H, W) for _ in range(N)]
-    nt = nestedtensor.nested_tensor(inputs)
-    def _dropout_nt():
-        torch.nn.functional.dropout(nt)
-    
-    return _dropout_nt
-
-#
-# interpolate
-#
-def interpolate_tensor_tensors_same_size():
-    inputs = [torch.randn(C, H, W) for _ in range(N)]
-    a = torch.stack(inputs)
-
-    def _interpolate_tensor():
-        torch.nn.functional.interpolate(a, inputs[0].unsqueeze(0).shape[-2])
-    return _interpolate_tensor
-
-def interpolate_nt_tensors_same_size():
-    inputs = [torch.randn(C, H, W) for _ in range(N)]
-    nt = nestedtensor.nested_tensor(inputs)
-    def _interpolate_nt():
-        torch.nn.functional.interpolate(nt)
-    
-    return _interpolate_nt
+class SegLayersBenchMark(object):
+    def __init__(self, args):
+        self.args = args
+        self.max_pool2d = torch.nn.MaxPool2d(kernel_size=(2, 2), stride=(2, 2), padding=(0, 0), dilation=(1, 1))
+        self.conv2d = torch.nn.Conv2d(args.C, 3, kernel_size= (1, 1), bias=False)
+        self.batch_norm = torch.nn.BatchNorm2d(args.C, 1e-05, 0.1)
+        self.batch_norm.eval()
 
+    def run(self):
+        for h_var in self.args.HV:  
```

Comment:
You can actually use https://docs.python.org/3.5/library/itertools.html#itertools.product for this